### ISSUE TITLE
Handle oversized tuple-valued wp.tid() dimensions

### ISSUE THREAD

**Issue by shi-eric** [NVIDIA/warp#1815]
Follow-up to #1802.

`wp.tid()` returns signed 32-bit thread coordinates. #1802 checks the retained leading extent for kernels that use scalar `wp.tid()`, but tuple-valued calls still have unhandled overflow cases.

The multidimensional launch bounds representation is relevant here:

```cpp
template <int N> struct launch_bounds_t {
    int shape[N];
    size_t size;
    size_t coord_mult;
};
```

`launch_coord()` uses the non-leading entries in `shape` for division and modulo. This leaves a few gaps:

- A kernel that unpacks `wp.tid()` into two or more coordinates, such as `i, j = wp.tid()`, but never calls scalar `wp.tid()`, does not trigger the scalar validation. A leading extent greater than `2**31` can therefore overflow the first returned coordinate.
- A non-leading extent of exactly `2**31` wraps when stored in `shape`. For `dim=(2, 2**31)`, linear index `2**31` should produce `(1, 0)`, but the current reconstruction produces `(0, -2147483648)`.
- When a kernel uses both scalar and tuple-valued `wp.tid()`, the existing scalar validation only covers the leading coordinate.

The total thread count is separate from the coordinate limits. A grid such as `(65536, 65536)` has more than `2**31` threads while each coordinate remains representable, so that case should continue to work.

The follow-up should define and enforce the tuple-valued limits consistently across direct, recorded, and JAX launch paths. Possible approaches include validating dimensions before packing them into the current bounds, changing the bounds representation, or combining the two. A larger representation would use more memory, so the implementation choice should remain open. The immediate goal is to prevent unsupported dimensions from silently producing incorrect coordinates.

**Comment by shi-eric** [comment]
This is addressed by 172ac26c9b4cbc0bf4e50ed1e3a56989a5f52934.

Tuple-only kernels and kernels mixing scalar and tuple-valued `wp.tid()` now validate every represented coordinate extent. Unsigned launch-shape storage preserves the valid `2**31` boundary, including `(2, 2**31)`, while larger represented extents are rejected. Grids such as `(65536, 65536)` remain supported because the limit applies to each coordinate extent, not the total thread count. Closing this issue as completed.

### Public API behavior

For nonempty launches, `wp.launch()` and recorded-command `set_dim()` reject represented extents above `2**31` with `ValueError`. JAX explicit and inferred launch dimensions use the same validation, including runtime revalidation after batching. The kernel-only `wp.tid()` builtin continues to return signed 32-bit coordinates.

APIC now writes format version 16 with unsigned launch extents. Readers accept versions 13–16 and reject ambiguous high-bit launch extents in older formats.

### Test coverage

- `warp/tests/test_template_launch_bounds.py`: coordinate limits across ranks 1–4, exact-boundary acceptance, large valid totals, empty launches, aliases, mixed scalar/tuple usage, compile-time-dead calls, folded dimensions, and recorded-command resizing on CPU/CUDA where applicable.
- `warp/tests/interop/test_jax.py`: explicit and inferred oversized tuple dimensions are rejected during tracing on CPU/CUDA; scalar rejection checks are updated for the shared diagnostic.
- `warp/tests/test_codegen.py`: direct tuple-valued calls infer kernel dimensionality without fallback reference scans.
- `warp/tests/test_apic.py`: writer-version checks on CPU/CUDA, plus CPU checks for legacy overflow rejection and current-format oversized launches without `wp.tid()`.

### Behavioral verification

All 133 selected committed tests passed. Separate targeted CPU/CUDA checks confirmed that `(2, 2**31)` reconstructs linear index `2**31` as `(1, 0)`, oversized tuple and mixed-arity launches are rejected through direct and recorded paths, and JAX rejects overflow introduced by `vmap`. Large-grid coordinate checks sampled native reconstruction without executing billions of threads.
